The animation control related changes

thesiv opened the pull request.

thesiv committed:
Support wxGenericAnimationCtrl by the animation control XRC handler

thesiv committed:
Fix an issue with the background in wxGenericAnimationCtrl

Static bitmap with alpha channel must also be drawn on the memory DC
instead of being used directly, just as the bitmaps with a mask.

vadz commented on src/xrc/xh_animatctrl.cpp:
```diff
@@ -31,16 +34,67 @@ wxAnimationCtrlXmlHandler::wxAnimationCtrlXmlHandler() : wxXmlResourceHandler()
+        ctrl->Hide();
 
-    wxScopedPtr<wxAnimation> animation(GetAnimation(wxT("animation")));
+    const wxString aniParam = "animation";
+    wxString path = GetParamValue(GetParamNode(aniParam));
+    if ( m_resource->GetFlags() & wxXRC_USE_ENVVARS )
+        path = wxExpandEnvVars(path);
```
What's wrong with using `GetFilePath()` here?

vadz commented on samples/xrc/myframe.cpp:
```diff
@@ -206,14 +207,17 @@ void MyFrame::OnAnimationCtrlPlay(wxCommandEvent& event)
 
     wxWindow *win = btn->GetParent();
     wxAnimationCtrl *ctrl = XRCCTRL(*win, "controls_animation_ctrl", wxAnimationCtrl);
+    wxGenericAnimationCtrl *generic =
+        XRCCTRL(*win, "controls_generic_animation_ctrl", wxGenericAnimationCtrl);
```
I'd probably take this inside `#ifdef wxHAS_NATIVE_ANIMATIONCTRL` as otherwise it's one and the same thing anyhow...

vadz reviewed (commented):
Thanks!

Just one small problem: I'd like to understand why do we duplicate `wxXmlResourceHandlerImpl::GetFilePath()` code instead of reusing it?

thesiv committed:
Add wxXmlResourceHandler::GetFilePath()

The function already implemented in wxXmlResourceHandlerImpl. Just make it
accessible from wxXmlResourceHandler.

thesiv commented:
Fixed.

vadz committed:
Get rid of src/xrc/xmladv.cpp file

It contained a single function which can be defined in xh_animatctrl.cpp
instead as it's only used there anyhow, and this file has no more reason
to exist, as "adv" library itself doesn't exist any longer.

vadz committed:
Add dummy wxBitmap::HasAlpha() to wxX11

This allows to write code using this function without any preprocessor
checks.

It should arguably be added to wxBitmapBase to ensure that it's
available in all ports, but should be done together with UseAlpha(),
which is not quite trivial, so leave it for later.

vadz committed:
Use GetAnimation() in wxAnimationCtrlXmlHandler again

This function basically only exists in order to be used in this handler,
so extend it to allow doing it by adding wxAnimationCtrlBase argument to
it, allowing it to create wxAnimation object compatible with the given
control.

This reduces code duplication and, incidentally, makes GetAnimation()
more useful for any user-defined XRC handlers.
